src/plugins/dashboard_embeddable_container/public/direct_access_link_generator.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import url from 'url';
+import { TimeRange, Filter, Query } from '../../data/public';
+import { setStateToKbnUrl } from '../../kibana_utils/public';
+import { DirectAccessLinkOptions } from '../../direct_access_links/public';
+
+export const STATE_STORAGE_KEY = '_a';
+export const GLOBAL_STATE_STORAGE_KEY = '_g';
+
+export const DASHBOARD_APP_LINK_GENERATOR = 'DASHBOARD_APP_LINK_GENERATOR';
+
+export interface DashboardAppLinkGeneratorState {
+  State: {
+    dashboardId?: string;
+    timeRange?: TimeRange;
+    filters?: Filter[];
+    query?: Query;
+    // If not given, will use the uiSettings configuration
+    useHash?: boolean;
+  };
+}
+
+export const createDirectAccessDashboardLinkGenerator = (
```

Comment:
I see code specific to Dasboard App here, and variables named "dashboard app":

![image](https://user-images.githubusercontent.com/9773803/75023378-01ca0380-5498-11ea-955b-57a4ce2e246a.png)

My understanding is that this code should be part of `dashboard` plugin, where Dasboard App will live and `dashboard_embeddable_container` should be just and independent implementation of Dashboard embeddable that can be rendered anywhere and should not be tied to the Dashboard App.

Reply:
Yea, I was talking to @majagrubic about this, should pull in @timroes too.  I think this plugin should be renamed `dashboard` and have all the dashboard app stuff when it migrates to the new plugin.  

> independent implementation of Dashboard embeddable that can be rendered anywhere and should not be tied to the Dashboard App.

I definitely see your point here (why I did it like this to begin with), but in the effort of minimizing our top level folders and grouping services by domain (and after the discussion in https://github.com/elastic/kibana/issues/47065), I think it makes sense that dashboard embeddable would be a service under dashboard app.  Users can still use feature controls to hide the app itself.  I think we want to move away though from having every embeddable implementation be its own unique top level plugin though.